### PR TITLE
treat autocontain settings as url patterns

### PR DESCRIPTION
**Pull request by arcnmx**
This aims to be more consistent and in line with the `seturl` family of settings.

I imagine this change will be inconvenient considering most users probably only care to specify an anchored domain pattern, but I take issue with it being inflexible. I think this change can be justified given that it's consistent with seturl/bindurl/etc, but an alternate approach may be to instead change `autocontain` excmd to include the `https?://etc` pattern, and still allow some other way (flag? `js tri.config.set`?) to set a full pattern when desired instead of hard-coding the pattern in `lib/autocontainers`.

EDIT: not sure what's with the test failure? as far as I can tell autocontain isn't used in any existing tests.

### CODE REVIEW

**Comment by glacambre** [comment]
Yeah, the e2e tests are flaky, you can ignore them :)

**Comment by bovine3dom** [comment]
We can use config update to stop this breaking settings for users:

https://github.com/tridactyl/tridactyl/blob/2f28555ac681db0b77daec03090070ac3d8014ac/src/lib/config.ts#L1302

Just take all autocontain domains and replace them with the `https?:// ...` regex.

I think it's important as breaking autocontain could lead to leaking information that users had actively tried to prevent leaking.

File: src/excmds.ts [hunk 1]
```diff
@@ -3316,20 +3316,16 @@ export function autocmd(event: string, url: string, ...excmd: string[]) {
...
 
-/** Automatically open a domain and all its subdomains in a specified container.
+/** Automatically open matching URLs in a specified container.
  *
  *  For declaring containers that do not yet exist, consider using `auconscreatecontainer true` in your tridactylrc.
  *  This allows tridactyl to automatically create containers from your autocontain directives. Note that they will be random icons and colors.
  *
  * ** NB: This is an experimental feature, if you encounter issues please create an issue on github. **
  *
- *  The domain is passed through as a regular expression so there are a few gotchas to be aware of:
```
Comment (bovine3dom) [inline src/excmds.ts]: It'd be handy if you put some examples here, e.g. one showing how to preserve the previous behaviour.

Comment (arcnmx) [reply src/excmds.ts]: Brought back the existing example, though it includes `google\,(co\.uk|com)` ... note the comma, I assume that's a typo?

**Comment by arcnmx** [comment]
> obviously those with RC files have no recourse

This may be a good argument to retain the existing behaviour for the `autocontain` command, and offer `autocontain -u ^https?://etc...` or something to adjust the configuration directly instead.

File: src/excmds.ts (hunk 1 above)
Comment (bovine3dom) [reply src/excmds.ts]: Yeah, should be a . :)

File: src/excmds.ts [hunk 2]
```diff
@@ -3316,25 +3316,41 @@ export function autocmd(event: string, url: string, ...excmd: string[]) {
     config.set("autocmds", event, url, excmd.join(" "))
 }
 
-/** Automatically open a domain and all its subdomains in a specified container.
+/** Automatically open matching URLs in a specified container.
+ *
+ *  This function accepts a `-u` flag to treat the pattern as a URL rather than a domain.
+ *  For example: `autocontain -u ^https?://[^/]*youtube\.com/ google` is equivalent to `autocontain youtube\.com google`
```
Comment (bovine3dom) [inline src/excmds.ts]: AFAICT this does the exact opposite of what it says: `-u` makes it treat the thing as a regex. `-r` for regex or maybe `-e` for extended mode makes more sense?

Comment (arcnmx) [reply src/excmds.ts]: As discussed on IRC the distinction here is "domain" vs "url", and the argument is always a regex. The lack of `-u` just means we wrap it in the `https?://[^/]<input>/` pattern so it doesn't match arbitrary URLs. Part of the issue here is that I changed the wording before adding the compatibility mode in, so I've reverted some of that to make the docs change more minimal which should help clear it up?

**Pull request merged**